File: scripts/extract_products.py
import json
import re
from pathlib import Path

BASE = Path(__file__).resolve().parents[1] / "public_html"

FILES = [
    ("Pickels.html", "Pickles"),
    ("Sweets.html", "Sweets"),
    ("Spices.html", "Spices"),
    ("Snacks.html", "Snacks"),
    ("Snacks3.html", "Snacks"),
    ("shopus3.html", "Mixed"),
]

pattern = re.compile(
    r"\{\s*name:\s*'([^']+)'\s*,\s*image:\s*'([^']+)'\s*,\s*stock:\s*'([^']+)'\s*,"
    r"\s*basePrice:\s*([0-9.]+)\s*,\s*ingredients:\s*'([^']*)'\s*,\s*benefits:\s*'([^']*)'\s*\}",
    re.DOTALL,
)

def extract_products():
    data = {}

    for file_name, category in FILES:
        content = (BASE / file_name).read_text(encoding="utf-8")
        block_match = re.search(r"const products\s*=\s*\[([\s\S]*?)\];", content)
        if not block_match:
            continue

        for match in pattern.findall(block_match.group(1)):
            name, image, stock, base_price, ingredients, benefits = match
            key = name.strip().lower()
            if key in data:
                continue

            base_val = float(base_price)
            data[key] = {
                "name": name.strip(),
                "image": image.strip(),
                "category": category,
                "stock": stock.strip(),
                "basePrice": base_val,
                "ingredients": ingredients.strip(),
                "benefits": benefits.strip(),
            }

    return list(data.values())


if __name__ == "__main__":
    products = extract_products()
    output_path = Path(__file__).with_name("products.json")
    with open(output_path, "w", encoding="utf-8") as fh:
        json.dump(products, fh, indent=2, ensure_ascii=False)
    print(f"Extracted {len(products)} products to {output_path}")

File: scripts/extractProducts.js
const fs = require('fs');
const path = require('path');

const baseDir = path.join(__dirname, '..', 'public_html');

const files = [
  { file: 'Pickels.html', category: 'Pickles' },
  { file: 'Sweets.html', category: 'Sweets' },
  { file: 'Spices.html', category: 'Spices' },
  { file: 'Snacks.html', category: 'Snacks' },
  { file: 'Snacks3.html', category: 'Snacks' },
  { file: 'shopus3.html', category: 'Mixed' },
];

const normalizeName = (name = '') => name.trim().toLowerCase();

const dataMap = new Map();

files.forEach(({ file, category }) => {
  const filePath = path.join(baseDir, file);
  const content = fs.readFileSync(filePath, 'utf8');
  const match = content.match(/const products\s*=\s*\[([\s\S]*?)\];/);
  if (!match) {
    console.warn(`No products array found in ${file}`);
    return;
  }

  const arrayLiteral = `[${match[1]}]`;
  let products;
  try {
    products = eval(arrayLiteral);
  } catch (err) {
    console.error(`Failed to parse products from ${file}:`, err.message);
    return;
  }

  products.forEach((product) => {
    if (!product || !product.name) {
      return;
    }

    const key = normalizeName(product.name);
    if (dataMap.has(key)) {
      return;
    }

    dataMap.set(key, {
      name: product.name.trim(),
      image: product.image,
      basePrice: product.basePrice,
      description: product.benefits || product.ingredients || '',
      ingredients: product.ingredients || '',
      category,
    });
  });
});

const combined = Array.from(dataMap.values());

const outputPath = path.join(__dirname, 'products.json');
fs.writeFileSync(outputPath, JSON.stringify(combined, null, 2), 'utf8');

console.log(`Extracted ${combined.length} unique products to ${outputPath}`);



